src/framework/global/io/path.cpp

```diff
@@ -213,16 +213,20 @@ bool mu::io::isAllowedFileName(const path_t& fn_)
     return true;
 }
 
-mu::io::path_t mu::io::escapeFileName(const mu::io::path_t& fn_)
+mu::io::path_t mu::io::escapeFileName(const mu::io::path_t& fn_, bool removeSpace)
 {
     //
     // special characters in filenames are a constant source
     // of trouble, this replaces some of them common in german:
     //
     String fn = fn_.toString();
-    fn = fn.simplified();
-    fn = fn.replace(' ',  '_');
     fn = fn.replace('\n', '_');
+    fn = fn.simplified();
```

Comment:
I think it's fine to call `simplified` first, and to remove `fn = fn.replace('\n', '_');`. Otherwise, newlines will be converted into `_` even if `removeSpace` is `false`, while it would be much more logical to convert newlines to spaces in that case.